weeks/week02/section02.md

```diff
@@ -0,0 +1,196 @@
+
+
+## 2.2 Reading
+
+Tuples are indexed in the same way as lists, meaning each element in the tuple can be accessed by its index. The index starts at 0, and negative indexing is also supported.
+
+
+### Syntax:
+
+```python
+test_data[0]  # First element
+test_data[1]  # Second element
+test_data[-1]  # Last element
+test_data[-2]  # Second-to-last element
+```
+
+
+### Using the `len()` function for tuples:
+
+Just like with lists, the `len()` function is used to find the number of elements in a tuple.
+
+
+#### Example:
+
+```python
+test_cases = ("Login Test", "Signup Test", "Logout Test")
+print(len(test_cases))  # Output: 3
+
+nested_tuple = ((1, 2, 3), (4, 5), (6,))
+print(len(nested_tuple))  # Output: 3
+```
+
+
+### Slicing Tuples:
+
+Tuples support `slicing`, similar to lists. Slicing allows extracting parts of a tuple without modifying the original data.
+
+
+#### Syntax:
+
+```python
+tuple_variable[start:stop:step]
+```
+
+
+#### Examples:
+
+- Basic slicing:
+
+```python
+numbers = (10, 20, 30, 40, 50, 60, 70, 80, 90, 100)
+first_three = numbers[:3]
+print(first_three)  # Output: (10, 20, 30)
+```
+
+- Slicing with steps:
+
+```python
+numbers = (10, 20, 30, 40, 50, 60, 70, 80, 90, 100)
+every_second = numbers[::2]
+print(every_second)  # Output: (10, 30, 50, 70, 90)
+```
+
+- Negative slicing
+
+```python
+numbers = (10, 20, 30, 40, 50)
+last_two = numbers[-2:]
+print(last_two)  # Output: (40, 50)
+```
+
+- Revert a tuple:
+
+```python
+numbers = (10, 20, 30, 40, 50)
+reverted_tuple = numbers[::-1]
+print(reverted_tuple)  # Output: (50, 40, 30, 20, 10)
+```
+
+
+- Out-of-Bounds indexing
+
+```python
+numbers = (10, 20, 30)
+out_of_bounds = numbers[2:10]
+print(out_of_bounds)  # Output: (30,)
+```
+
+
+## 2.3 Updating
+
+Tuples are `immutable`, meaning they cannot be changed once created. You `cannot` update the elements of a tuple after it has been defined. With that said, there is a workaround for "updating" a tuple.
+
+If you need to "update" a tuple, you can convert it into a list, make the necessary changes, and then convert it back into a tuple. This is common in test automation when dealing with static test data that may need minor adjustments.
+
+
+### Example:
+
+```python
+test_data = ("Login", "Signup", "Logout")
+
+# Convert tuple to list
+test_data_list = list(test_data)
+test_data_list[0] = "Login Test"  # Update the first element
+
+# Convert back to tuple
+test_data = tuple(test_data_list)
+print(test_data)  # Output: ('Login Test', 'Signup', 'Logout')
+```
+
+
+## 2.4 Deleting
+
+Tuples, being immutable, cannot have their elements deleted directly. However, you can delete the entire tuple using `del` statement.
+
+### Syntax:
+
+```python
+del tuple_name
+```
+
+
+
+## Practical Exercises (Drills):
+
+1. Create a tuple containing multiple types of elements, including an integer, a boolean, a list, another tuple, and a string.
```

Comment:
Are there any [TI] questions that we can add here?
cc @ericrommel

Reply:
Hmm. Good question. I don't have any regarding tuple other than comparison with Lists or Sets. Feel free to add add/remove here.